src/companies/tests/test_master_procedure_viewset.py

```diff
@@ -0,0 +1,206 @@
+
+
+def test_master_procedure_detail(reader_client: ApiClient, master_procedure: MasterProcedure):
+    assert reader_client.get(master_procedure.get_absolute_url()) == MasterProcedureSerializer(master_procedure).data  # type: ignore[no-untyped-call]
+
+
+def test_point_managing_staff_can_update_master_procedure(
+    as_point_managing_staff: ApiClient,
+    master_procedure: Procedure,
+    procedure: Procedure,
+    employee: Employee,
+    master_procedure_data: dict[str, Any],
+    assert_master_procedure: ModelAssertion,
+):
+    response = as_point_managing_staff.put(master_procedure.get_absolute_url(), master_procedure_data)  # type: ignore[no-untyped-call]
+    master_procedure.refresh_from_db()
+
+    assert_master_procedure(master_procedure_data, procedure=procedure, employee=employee)
+    assert response == MasterProcedureSerializer(MasterProcedure.objects.get(**master_procedure_data)).data
+
+
+def test_point_non_managing_staff_cannot_update_master_procedure(
+    as_point_non_managing_staff: StatusApiClient,
+    master_procedure: MasterProcedure,
+    master_procedure_data: dict[str, Any],
+):
+    as_point_non_managing_staff.put(  # type: ignore[no-untyped-call]
+        master_procedure.get_absolute_url(),
+        master_procedure_data,
+        expected_status=as_point_non_managing_staff.expected_status,
+    )
+
+
+@pytest.mark.parametrize("invalid_data", [{"coef": -0.1}, {"coef": 1.1}])
+def test_update_master_procedure_invalid_data(
+    as_point_managing_staff: ApiClient,
+    master_procedure: MasterProcedure,
+    factory: FixtureFactory,
+    employee: Employee,
+    procedure: Procedure,
+    invalid_data: dict[str, str],
+):
+    as_point_managing_staff.put(  # type: ignore[no-untyped-call]
+        master_procedure.get_absolute_url(),
+        factory.master_procedure_data(employee=employee.id, procedure=procedure.id, **invalid_data),
+        expected_status=status.HTTP_400_BAD_REQUEST,
+    )
+
+
+@freeze_time()
+def test_point_managing_staff_can_delete_master_procedure(
+    as_point_managing_staff: ApiClient,
+    master_procedure: MasterProcedure,
+    assert_doesnt_exist: ExistCheckAssertion,
+):
+    as_point_managing_staff.delete(master_procedure.get_absolute_url())  # type: ignore[no-untyped-call]
+
+    assert_doesnt_exist(MasterProcedure)
+    assert MasterProcedure.include_archived.filter(archived=timezone.now()).exists()
+
+
+@freeze_time()
+def test_point_non_managing_staff_cannot_delete_procedure(
+    as_point_non_managing_staff: StatusApiClient,
+    master_procedure: MasterProcedure,
+    assert_exists: ExistCheckAssertion,
+):
+    as_point_non_managing_staff.delete(  # type: ignore[no-untyped-call]
+        master_procedure.get_absolute_url(),
+        expected_status=as_point_non_managing_staff.expected_status,
+    )
+
+    assert_exists(MasterProcedure)
+    assert not MasterProcedure.include_archived.filter(archived=timezone.now()).exists()
```

Comment:
Current check doesn't guarantee that master procedure was not deleted at all, it's better to check existance this way.

```suggestion
    assert MasterProcedure.include_archived.filter(id=master_procedure.id).exists()
```